src/TLuaInterpreter.cpp

```diff
@@ -1015,16 +1015,26 @@ int TLuaInterpreter::setMiniConsoleFontSize(lua_State* L)
 int TLuaInterpreter::getLineNumber(lua_State* L)
 {
     Host& host = getHostFromLua(L);
-    if (lua_isstring(L, 1)) {
-        string window = lua_tostring(L, 1);
-        QString _window = window.c_str();
-        lua_pushnumber(L, mudlet::self()->getLineNumber(&host, _window));
+    QString windowName;
+    int s = 0;
+
+    if (lua_gettop(L) > 1) { // Have more than one argument so first must be a console name
+        if (!lua_isstring(L, ++s)) {
+            lua_pushfstring(L, "getLineNumber: bad argument #%d type (window name as string expected, got %s!)", s, luaL_typename(L, s));
+            lua_error(L);
+            return 1;
+        } else {
+            windowName = QString::fromUtf8(lua_tostring(L, s));
+        }
+    }
+
+    if (windowName.isEmpty() || windowName.compare(QStringLiteral("main"), Qt::CaseSensitive) == 0) {
+        lua_pushnumber(L, host.mpConsole->getLineNumber());
         return 1;
     } else {
-        lua_pushnumber(L, host.mpConsole->getLineNumber());
+        lua_pushnumber(L, mudlet::self()->getLineNumber(&host, windowName));
```

Comment:
:bulb: Off the top of my head - what happens if `windowName` does not exist as a `TConsole` instance - some of the functions that target *all* consoles rather than just the **main** one do produce a run-time **nil+error message** in that situation do they not?

Reply:
The original function does not report it - but I'll look into upgrading it to do so for consistency. Good point.

Reply:
As part of this PR?

Reply:
Yeah.